### DISCUSSION TITLE
Traces of a cancelled booking

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#208] [General]
Hi Franky,
Is there any way of tracing the cancellation of a booking (who cancelled it and when). Normally, only an administrator or the person concerned can cancel a booking. I need to find this information.
Thanks, have a nice day.
Frédéric

**Comment by liedekef** [comment]
Well, normally you can see in EME when the booking was cancelled if a mail was sent (in the EME mailings overview).
Based on that timestamp, you'll need to then check the webserver logfiles for the IP of the person who did it. Problem is: if the person itself cancelled his booking, he doesn't even need to be logged in. Search for "eme_cancel" in the webserver logfiles.
Also: cancel of a booking is normally only done by the person himself (unless an admin knew the cancel url). In the EME backend you can trash a booking, but another mail is sent for that. A booking in trash can also be recovered of course.

**Comment by f14m07** [comment]
Thanks Franky

**Comment by liedekef** [comment]
I know it is not much to go on. I already thought about implementing full auditing, but that requires a DB table that can get large pretty fast. Maybe I'll think about another method for auditing ... any hints are welcome.

**Comment by f14m07** [comment]
You're right. I've got tools that log a lot of things, but you can't do it over too long a period.
